websockets/exceptions.py

```diff
@@ -26,13 +26,31 @@ class InvalidOrigin(InvalidHandshake):
     """
 
 
+class InvalidStatus(InvalidHandshake):
+    """
+    Exception raised when a handshake response status code is invalid.
+
+    Provides the integer status code and reason in its ``code`` and
+    ``reason`` attributes, respectively.
+
+    """
+    def __init__(self, code, reason=None):
+        self.code = code
+        self.reason = reason
+        message = 'Bad status code: {}'.format(code)
+        if reason:
+            message += ' ({})'.format(reason)
+        super().__init__(message)
+
+
 class InvalidState(Exception):
     """
     Exception raised when an operation is forbidden in the current state.
 
     """
 
 
+# TODO: DRY up with InvalidStatus?
```

Comment:
`InvalidStatus` stores a HTTP status code (three digits). `ConnectionClosed` stores a websocket status code (four digits). That makes them different types — even if we represent both as integers because we don't bother too much with types in Python. For this reason, I don't think we should merge these two exceptions types.